Assets/Scripts/Entity/Unit/Unit.cs

```diff
@@ -37,5 +63,43 @@ public void onClick()
             GameManager.instance.selectedUnit = gameObject;
         }
     }
+    
+    /* Moves the unit to a point */
+    public void moveTo(Vector3 point) 
+    {
+        gameObject.GetComponent<NavMeshAgent>().SetDestination(point);
+    }
+
+    /* Sets attacking flags and chases an enemy to attack */
+    public void startAttacking(GameObject entity)
+    {
+        targettedEntity = entity;
+        isAttacking = true;
+    }
+
+    /* Performs an attack against the selected entity */
+    public void performAttack(Unit target)
+    {
+        bool isTargetAlive = target.takeDamage(unitEntity.damage);
+        attackCooldown = unitEntity.attackTime;
 
+        if (!isTargetAlive)
+        {
+            targettedEntity = null;
```

Comment:
Maybe delegate this function to the game manager, or some manager that keeps track of the objects in the scene.